### PR TITLE
BUG: Index.getitem returning wrong result with negative step for arrow

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

```
idx = Index(list("bcdxy"), dtype="string[pyarrow_numpy]")
idx[slice(4, -6, -1)]
```

This omits b for arrow dtypes and looks like a bug on the arrow side, cc @jorisvandenbossche

### CODE REVIEW

File: pandas/core/arrays/arrow/array.py [hunk 1]
```diff
@@ -553,6 +553,13 @@ def __getitem__(self, item: PositionalIndexer):
             )
         # We are not an array indexer, so maybe e.g. a slice or integer
         # indexer. We dispatch to pyarrow.
+        if isinstance(item, slice):
+            if item.start == item.stop:
+                pass
+            elif item.start == -len(self) - 1:
```
Comment (lukemanley) [inline pandas/core/arrays/arrow/array.py:559]: should this be `item.start <= -len(self)`?

Not sure if you want to mimic numpy behavior here but numpy allows slicing beyond the len of the array:

```
In [1]: import numpy as np

In [2]: np.array(["a", "b", "c"])[slice(-10, 10)]
Out[2]: array(['a', 'b', 'c'], dtype='<U1')
```

Comment (phofl) [reply pandas/core/arrays/arrow/array.py]: Yeah that makes sense, updated

File: pandas/core/arrays/arrow/array.py [hunk 2]
```diff
@@ -553,6 +553,17 @@ def __getitem__(self, item: PositionalIndexer):
...
         # indexer. We dispatch to pyarrow.
+        if isinstance(item, slice):
+            if item.start == item.stop:
+                pass
+            elif (
+                item.stop is not None
+                and item.stop < -len(self)
+                and item.step is not None
+                and item.step < 0
+            ):
```
Comment (jorisvandenbossche) [inline pandas/core/arrays/arrow/array.py:564]: Can you add a comment about why we do this (or refer to a pyarrow bug report or this PR)

**Comment by jorisvandenbossche** [comment]
Opened an issue on the Arrow side -> https://github.com/apache/arrow/issues/38768

File: pandas/core/arrays/arrow/array.py (hunk 2 above)
Comment (phofl) [reply pandas/core/arrays/arrow/array.py:565]: added the arrow ref

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @phofl

**Pull request merged**